release-content/0.15/migration-guides/14082_AssetReader_read_now_returns_an_opaque_type.md

```diff
@@ -0,0 +1,38 @@
+The trait method `bevy_asset::io::AssetReader::read` (and `read_meta`) now return an opaque type instead of a boxed trait object. Implementors of these methods should change the type signatures appropriately
+
+```rust
+impl AssetReader for MyReader {
+    // Before
+    async fn read<'a>(&'a self, path: &'a Path) -> Result<Box<Reader<'a>>, AssetReaderError> {
+        let reader = // construct a reader
+        Box::new(reader) as Box<Reader<'a>>
+    }
+
+    // After
+    async fn read<'a>(&'a self, path: &'a Path) -> Result<impl Reader + 'a, AssetReaderError> {
+        // create a reader
+    }
+}
+```
+
+`bevy::asset::io::Reader` is now a trait, rather than a type alias for a trait object. Implementors of `AssetLoader::load` will need to adjust the method signature accordingly
```

Comment:
```suggestion
`bevy::asset::io::Reader` is now a trait, rather than a type alias for a trait object. Implementors of `AssetLoader::load` will need to adjust the method signature accordingly:
```